scss/_spinners.scss

```diff
@@ -0,0 +1,83 @@
+//
+// Rotating border
+//
+
+@keyframes spinner-border {
+  to { transform: rotate(360deg); }
+}
+
+.spinner-border {
+  position: relative;
+  display: inline-block;
+  width: 2rem;
+  height: 2rem;
+  overflow: hidden;
+  text-indent: -999em;
+  vertical-align: text-bottom;
+  border: .25em solid;
+  border-color: currentColor transparent currentColor currentColor;
+  border-radius: 50%;
+  animation: spinner-border .75s linear infinite;
+}
+
+.spinner-border-reverse {
+  border-color: transparent currentColor transparent transparent;
+
+  &::after {
+    position: absolute;
+    top: -.25em;
+    right: -.25em;
+    bottom: -.25em;
+    left: -.25em;
+    display: inline-block;
+    content: "";
+    border: .25em solid rgba(0, 0, 0, .1);
+    border-radius: 50%;
+  }
+}
+
+.spinner-border-sm {
+  width: 1rem;
+  height: 1rem;
+  border-width: .2em;
+
+  &.spinner-border-reverse::after {
+    border-width: .2em;
+  }
+}
+
+//
+// Growing circle
+//
+
+@keyframes spinner-grow {
+  0% {
+    opacity: 0;
+    transform: scale(0);
+  }
+  50% {
+    opacity: 1;
+  }
+  100% {
+    opacity: 0;
+    transform: scale(1);
+  }
+}
+
+.spinner-grow {
```

Comment:
`.spinner-border` and `.spinner-grow` share equal properties like `display`, `position`, `width` and `height` and so on.

We could apply nesting via SCSS operators to keep source file smaller and reduce (potential) future maintainance effort.

Reply:
We don't really use placeholders or extends in Bootstrap, so I'm not inclined to see an abstraction here. I also don't want to add a new base class, or a shared ruleset, because that increases the number of selectors. Open to improvements in the future, but figured I'd explain that quick here.